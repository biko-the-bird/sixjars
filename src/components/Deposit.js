import React, {Component} from 'react';
import {connect} from 'react-redux';
import { setJars } from "../redux/actions/balance";
import { updateJars } from "../helpers";

export class Deposit extends Component {

    constructor(props) {
        super(props);
        this.state = {
            deposit: 0
        }
    }

    handleDeposit = () => {
        const jars = updateJars(this.state.jars, this.props.jars);
        this.props.setJars(jars);
        this.setState({deposit: 0});
    }

    render() {
        return (
            <div>
                  <label>Deposit</label>
          <input
            type="number"
            className="deposit-input"
            value={this.state.deposit}
            onChange={e => this.setState({ deposit: e.target.value })}
          />
          <button className="deposit-btn" onClick={() => this.handleDeposit()}>
            Deposit
          </button>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {jars: state}
}

export default connect(mapStateToProps,{setJars})(Deposit);