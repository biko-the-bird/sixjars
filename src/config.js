
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//this is a configuration for enzyme that needs
//to be imported in each testing file
const config = configure({ adapter: new Adapter() });
export default config;