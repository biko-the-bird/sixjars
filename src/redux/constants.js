export const SET_JARS = "SET_JARS";
export const WITHDRAW_JAR = "WITHDRAW_JAR";
export const JARS_COOKIE = "JARS_COOKIE";
export const NEC = "NEC";
export const FFA = "FFA";
export const EDU = "EDU";
export const GIV = "GIV";
export const LTS = "LTS";
export const PLA = "PLA";


export const INIT_JARS = {NEC: 0, FFA: 0,GIV: 0, PLA: 0,EDU: 0,LTS: 0};
export const JAR_DESC = {NEC: "Absolute neccesites only. i.e. Food, housing, transportation, clothes.", FFA: "Do not spend this account. You can only invest it.", GIV: "To be donated to things you want to support. i.e. homeless, buying a friends mixtape", PLA: "guilt free, spend on anything you want", EDU: "for classes or gym memberships.", LTS: "Savings for big purchases. i.e. a car"}