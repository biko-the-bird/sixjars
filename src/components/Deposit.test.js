import React from 'react';
import {shallow} from 'enzyme';
import config from '../config';
import {Deposit} from './Deposit';
import * as constants from '../redux/constants';

describe("deposit", () => {
    const mockSetJars = jest.fn();
    const props = {setJars: mockSetJars, jars: constants.INIT_JARS};
    const deposit = shallow(<Deposit {...props}/>);
    it("renders", () => {
        expect(deposit).toMatchSnapshot();
    })


    describe('user adds a deposit', () => {

        var newDeposit = 10;

        beforeEach(() => {
            deposit.find('.deposit-input').simulate('change',{target: {value: newDeposit}});
        });
        it("saves deposit that user inputs to local `state`", () => {
            expect(deposit.state().deposit).toEqual(newDeposit);
        })

        describe('user submits a new deposit', () => {
            beforeEach(() => {
                deposit.find('.deposit-btn').simulate('click');
            });

            it('triggers the `setJars()` method received from props', () => {
                expect(mockSetJars).toHaveBeenCalled();
            })
            
           

            it("clears the local state", () => {
                expect(deposit.state().deposit).toEqual(0);
            });
        })
       
    });
})