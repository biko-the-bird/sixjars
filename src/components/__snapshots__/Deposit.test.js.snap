// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deposit renders 1`] = `ShallowWrapper {}`;
